tests/services/AchievementRuntime_Tests.cpp

```diff
@@ -180,8 +182,8 @@ TEST_CLASS(AchievementRuntime_Tests)
         runtime.mockGameContext.AddAchivement(3U, "1=1.10.");
         runtime.mockGameContext.AddAchivement(5U, "1=1.2.");
 
-        auto* pAchievement3 = runtime.mockGameContext.FindAchievement(3U);
-        auto* pAchievement5 = runtime.mockGameContext.FindAchievement(5U);
+        const auto pAchievement3 = gsl::make_not_null(runtime.mockGameContext.FindAchievement(3U));
+        const auto pAchievement5 = gsl::make_not_null(runtime.mockGameContext.FindAchievement(5U));
```

Comment:
Why remove the * from auto? It makes it more clear what the variable type is. It also prevents possible problems if the implicit type changes to not be a pointer anymore. The compiler will automatically call the copy constructor, and as long as the -> operator is still valid, you won't even know anything has changed.

Reply:
auto is relpaced with the type name to show the actual class is not a pointer, it just wraps a pointer to enforce it can't be null.